src/VocabularyManager.Core/Interfaces/IStateManager.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace VocabularyManager.Core.Interfaces
+{
+    public interface IStateManager
```

Comment:
this is what I will write it's very debatable. I think Interfaces should not be in `Core` but in `Application` (in your case `UseCases`) 
Core should contain the business logic and domain entities. Interfaces are not typically placed directly in the core layer, as interfaces define contracts for interactions between different parts of the system, rather than containing business rules themselves.